### PR TITLE
Support for certificates with expiration date

### PR DESCRIPTION
**Pull request by poszu**
Support for certificates with an optional expiration date. A certificate is a scale-encoded combination of the node's ID and (optional) expiration time. If the expiration time is `None`, the certificate does not expire.

The certifier will create expiring certificates if the expiration duration is set in the config.

### CODE REVIEW

File: certifier/src/certifier.rs [hunk 1]
```diff
@@ -20,71 +23,240 @@ pub struct CertifyRequest {
...
+    pub pub_key: Vec<u8>,
+}
+
+#[derive(Debug, Decode, Encode)]
+pub struct Certificate {
+    // ID of the node being certified
+    pub pub_key: Vec<u8>,
+    /// Unix timestamp
+    pub expiration: Option<Compact<u64>>,
 }
```
Comment (fasmat) [inline certifier/src/certifier.rs:45]: Maybe a bit late, but we could have gone the route of using actual x509 certificates and then use client authentication (mTLS) against the certifier, would have probably been less implementation effort and made it easier to develop 3rd party tools or alternative implementations.

File: certifier/src/certifier.rs [hunk 2]
```diff
@@ -20,71 +23,240 @@ pub struct CertifyRequest {
...
+        Ok(result) => {
+            let response = CertifyResponse {
+                certificate: result.0.to_vec(),
+                signature: result.1.to_vec(),
+                pub_key: state.signer.verifying_key().to_bytes().to_vec(),
+            };
+            Ok(Json(response))
+        }
+        Err(e) => {
+            return Err((StatusCode::FORBIDDEN, format!("invalid proof: {e:?}")));
```
Comment (fasmat) [inline certifier/src/certifier.rs:75]: I assume this should return an error like "not authorized" instead?

**Review by fasmat — APPROVED** [review]
There is a linter warning in `profiler/src/main.rs` that maybe can be fixed in this PR as well?

File: certifier/src/certifier.rs (hunk 2 above)
Comment (poszu) [reply certifier/src/certifier.rs:75]: Wdym? The `format!("invalid proof: {e:?}")` part is put in the response body and informs the caller why he was refused.

File: certifier/src/certifier.rs (hunk 1 above)
Comment (poszu) [reply certifier/src/certifier.rs:45]: Could a x509 certificate be only valid for a single nodeID? A certificate should allow for only 1 poet registration per round.

**Comment by poszu** [comment]
> There is a linter warning in `profiler/src/main.rs` that maybe can be fixed in this PR as well?

That's already fixed on the `main` branch :)

File: certifier/src/certifier.rs (hunk 1 above)
Comment (fasmat) [reply certifier/src/certifier.rs:45]: Yes, a certificate would be for a specific public key until a given expiration date. I don't think the effort to change it now is worth it though.

**Review by fasmat — APPROVED** [review]

**Pull request merged**